### ISSUE TITLE
GCE- Clouddriver errors out while caching GoogleZonalServerGroupCachingAgent

### ISSUE THREAD

**Issue by sweetib** [spinnaker/spinnaker#2155]
*Title*
Clouddriver errors out while caching GoogleZonalServerGroupCachingAgent

*Cloud Provider*
GCP

*Environment*
I am running Spinnaker on GCE Ubuntu 16.04.3 with clouddriver version **1.743.0**.

*Description*
After updating the clouddriver I am observing below errors. I even tried to downgrade it to version `1.741.0` the error still persist. My older version of clouddriver(`1.669.3`) is running fine without any issues. 

Can anyone help me with latest working version of clouddriver on GCE?

*Steps to Reproduce*

Update the spinnaker clouddriver on GCE to latest version. Add GCP account having some regional and zonal server groups.

*Clouddriver Error Logs*
```
groovy.lang.GroovyRuntimeException: Ambiguous method overloading for method java.util.ArrayList#addAll.
Cannot resolve which method to invoke for [null] due to overlapping prototypes between:
	[interface java.util.Collection]
	[interface java.lang.Iterable]
	[interface java.util.Iterator]
	at groovy.lang.MetaClassImpl.chooseMostSpecificParams(MetaClassImpl.java:3263) ~[groovy-all-2.4.12.jar:2.4.12]
	at groovy.lang.MetaClassImpl.chooseMethodInternal(MetaClassImpl.java:3216) ~[groovy-all-2.4.12.jar:2.4.12]
	at groovy.lang.MetaClassImpl.chooseMethod(MetaClassImpl.java:3159) ~[groovy-all-2.4.12.jar:2.4.12]
	at groovy.lang.MetaClassImpl.getMethodWithCachingInternal(MetaClassImpl.java:1331) ~[groovy-all-2.4.12.jar:2.4.12]
	at groovy.lang.MetaClassImpl.createPojoCallSite(MetaClassImpl.java:3391) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createPojoSite(CallSiteArray.java:132) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:166) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) ~[groovy-all-2.4.12.jar:2.4.12]
	at java_util_List$addAll$11.call(Unknown Source) ~[na:na]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent.fetchInstanceTemplates(GoogleZonalServerGroupCachingAgent.groovy:180) ~[clouddriver-google-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent$fetchInstanceTemplates$11.callStatic(Unknown Source) ~[na:na]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent.constructServerGroups(GoogleZonalServerGroupCachingAgent.groovy:140) ~[clouddriver-google-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent.constructServerGroups(GoogleZonalServerGroupCachingAgent.groovy) ~[clouddriver-google-1.741.0.jar:1.741.0]
	at sun.reflect.GeneratedMethodAccessor257.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:166) ~[groovy-all-2.4.12.jar:2.4.12]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent.getServerGroups(GoogleZonalServerGroupCachingAgent.groovy:124) ~[clouddriver-google-1.741.0.jar:1.741.0]
	at sun.reflect.GeneratedMethodAccessor256.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59) ~[groovy-all-2.4.12.jar:2.4.12]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:166) ~[groovy-all-2.4.12.jar:2.4.12]
	at com.netflix.spinnaker.clouddriver.google.provider.agent.GoogleZonalServerGroupCachingAgent.loadData(GoogleZonalServerGroupCachingAgent.groovy:99) ~[clouddriver-google-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgentWithoutStore(CachingAgent.java:79) ~[cats-core-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgent(CachingAgent.java:72) ~[cats-core-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentExecutionAction.execute(ClusteredAgentScheduler.java:272) ~[cats-redis-1.741.0.jar:1.741.0]
	at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentJob.run(ClusteredAgentScheduler.java:246) ~[cats-redis-1.741.0.jar:1.741.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_151]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
```

**Comment by lwander** [comment]
Hey @jtk54 could you take a look at this?

**Comment by jtk54** [comment]
Reproduced, fix incoming.

**Comment by jtk54** [comment]
Merged https://github.com/spinnaker/clouddriver/pull/2216.

**Comment by jtk54** [comment]
Released as Clouddriver v1.744.0. Closing, feel free to reopen if this persists.

**Comment by sweetib** [comment]
@jtk54 Thanks for the quick fix. Meanwhile I tried updating spinnaker today and facing issues which I have raised here https://github.com/spinnaker/spinnaker/issues/2172.
